text/2020-03-11-versioned-kv.md

```diff
@@ -0,0 +1,107 @@
+# Versioned KV (VerKV)
+
+## Summary
+
+This proposal presents another KV interface - Versioned KV (VerKV), a multi-versioned KV between RawKV and TxnKV.
+
+## Motivation
+
+At present, Yidian Zixun adopts TiKV RawKV for Key-Value storage. Due to performance concerns, TxnKV is not used. However, RawKV only provides simple KV semantics, and does not support multiple versions based on timestamp, which causes some problems: 
+  - When services are migrated from HBase to TiKV, RawKV does not fulfil some major functionality requirements such as data expiration and historical data version;
+  - When developing backup and synchronization functions based on RawKV, it is difficult to solve the problems of online data migration, incremental backup, and data synchronization between two data centers because timestamp is not supported.
+
+In order to solve the above problems, this proposal presents another kv interface - VerKV, which mainly adds version information such as timestamp to the key. At the same time, this interface will be designed to be approximate to the semantics of HBase, which will facilitate the service migration from HBase to TiKV, thereby integrating TiKV better into the HBase ecosystem.
+
+## Detailed designs
+
+### Functions and scenarios
```

Comment:
Do we need to support multiple cfs or just one cf

Reply:
RawKV supports get/put with cf, but restrict the scope with [defaut, write, lock]. Maybe VerKV should not support multiple cfs, and only operate on its dedicated cf?  /cc @zhangjinpeng1987 @BusyJay

Reply:
I thought raw kv lets you write to any CF?

Reply:
To be consist with RawKV，here just write one cf "verkv".